jobspec/parse.go

```diff
@@ -452,6 +462,75 @@ func parseTasks(result *[]*structs.Task, list *ast.ObjectList) error {
 	return nil
 }
 
+func parseServices(jobName string, taskGroupName string, task *structs.Task, serviceObjs *ast.ObjectList) error {
+	task.Services = make([]structs.Service, len(serviceObjs.Items))
+	for idx, o := range serviceObjs.Items {
+		var service structs.Service
+		var m map[string]interface{}
+		if err := hcl.DecodeObject(&m, o.Val); err != nil {
+			return err
+		}
+
+		delete(m, "check")
+
+		if err := mapstructure.WeakDecode(m, &service); err != nil {
+			return err
+		}
+
+		if idx > 0 && service.Name == "" {
+			return fmt.Errorf("More than one service block is declared, please name each service explicitly")
+		}
+
+		if service.Name == "" {
+			service.Name = fmt.Sprintf("%s-%s-%s", jobName, taskGroupName, task.Name)
+		}
```

Comment:
So we are always forcing the jobname-group-task-name on a service name? I don't mind that, I was wondering user would know what he is doing when he is explicitly name a service.